app/src/main/java/com/quran/labs/androidquran/ui/helpers/QuranDisplayHelper.java

```diff
@@ -73,6 +73,29 @@ public static long displayMarkerPopup(Context context, int page,
     return System.currentTimeMillis();
   }
 
+  // same logic used in displayMarkerPopup method
+  public static String displayRub3(Context context, int page){
+    int rub3 = QuranInfo.getRub3FromPage(page);
+    int hizb = (rub3 / 4) + 1;
+    StringBuilder sb = new StringBuilder();
+    sb.append(" , ");
+    if (rub3 == -1) {
+      return "";
+    }
+    int remainder = rub3 % 4;
```

Comment:
For the sake of readability (just a suggestion):

    switch(rub3%4) {
        case 1:
            sb.append(context.getString(R.string.quran_rob3)).append(' ');
            break;
        case 2:
            sb.append(context.getString(R.string.quran_nos)).append(' ');
            break;
        case 3:
            sb.append(context.getString(R.string.quran_talt_arb3)).append(' ');
            break;
        default: // case 0
            // do nothing
    }